compose/ui/ui/src/webCommonW3C/kotlin/androidx/compose/ui/window/ComposeWindow.js.kt

```diff
@@ -134,75 +174,80 @@ private class ComposeWindow(
+        addTypedEvent<MouseEvent>("mousedown") { event ->
             isPointerPressed = true
-            skikoView.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.DOWN))
+            layer.view.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.DOWN))
         }
 
-        canvas.addTypedEvent<MouseEvent>("mouseup") { event, skikoView ->
+        addTypedEvent<MouseEvent>("mouseup") { event ->
             isPointerPressed = false
-            skikoView.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.UP))
+            layer.view.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.UP))
         }
 
-        canvas.addTypedEvent<MouseEvent>("mousemove") { event, skikoView ->
+        addTypedEvent<MouseEvent>("mousemove") { event ->
             if (isPointerPressed) {
-                skikoView.onPointerEvent(event.toSkikoDragEvent())
+                layer.view.onPointerEvent(event.toSkikoDragEvent())
             } else {
-                skikoView.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.MOVE))
+                layer.view.onPointerEvent(event.toSkikoEvent(SkikoPointerEventKind.MOVE))
             }
         }
 
-        canvas.addTypedEvent<WheelEvent>("wheel") { event, skikoView ->
-            skikoView.onPointerEvent(event.toSkikoScrollEvent())
+        addTypedEvent<WheelEvent>("wheel") { event ->
+            layer.view.onPointerEvent(event.toSkikoScrollEvent())
         }
 
         canvas.addEventListener("contextmenu", { event ->
             event.preventDefault()
         })
 
-        canvas.addTypedEvent<KeyboardEvent>("keydown") { event, skikoView ->
-            val processed = skikoView.onKeyboardEventWithResult(event.toSkikoEvent(SkikoKeyboardEventKind.DOWN))
+        addTypedEvent<KeyboardEvent>("keydown") { event ->
+            val processed = layer.view.onKeyboardEventWithResult(event.toSkikoEvent(SkikoKeyboardEventKind.DOWN))
             if (processed) event.preventDefault()
         }
 
-        canvas.addTypedEvent<KeyboardEvent>("keyup") { event, skikoView ->
-            val processed = skikoView.onKeyboardEventWithResult(event.toSkikoEvent(SkikoKeyboardEventKind.UP))
+        addTypedEvent<KeyboardEvent>("keyup") { event ->
+            val processed = layer.view.onKeyboardEventWithResult(event.toSkikoEvent(SkikoKeyboardEventKind.UP))
             if (processed) event.preventDefault()
         }
-
-        initMediaEventListener {
-            resize(getParentContainerBox())
-        }
     }
 
     init {
         initEvents(canvas)
+        state.init()
 
         canvas.setAttribute("tabindex", "0")
 
         layer.setDensity(density)
         layer.setContent {
             CompositionLocalProvider(
                 LocalSystemTheme provides systemThemeObserver.currentSystemTheme.value,
-                content = content
+                content = {
+                    content()
+                    rememberCoroutineScope().launch {
```

Comment:
I'm okay with using rememberCoroutineScope() here instead of LaunchedEffect. In this case both can be used just fine AFAIK.
What was the reason you preferred to use rememberCoroutineScope instead of LaunchedEffect?

Reply:
I've decided to do it after following comment:
https://github.com/JetBrains/compose-multiplatform-core/pull/1166#discussion_r1514790462